csrc/cutlass_extensions/vllm_numeric_conversion.cuh

```diff
@@ -140,6 +149,142 @@ struct ArrayConverterPacked32Bit {
   }
 };
 
+// for Array<int8_t, N> <= Array<vllm_uint4b8_t, N>
+template <FloatRoundStyle Round, int N>
+struct NumericArrayConverter<int8_t, vllm_uint4b8_t, N, Round> {
+  using result_type = Array<int8_t, N>;
+  using source_type = Array<vllm_uint4b8_t, N>;
+
+  static FloatRoundStyle const round_style = Round;
+
+ private:
+  struct RegConvert {
+    template <typename PackedResultType>
+    CUTLASS_DEVICE static PackedResultType convert(Array<uint32_t, 1> src_) {
+      cutlass::AlignedArray<uint32_t, std::max(PackedResultType::kElements / 4,
+                                               size_t(1))>
+          r;
+      uint32_t src = src_[0];
+      // Determines if to get from the signed or unsigned candidates
+      uint32_t sign = (src & 0x88888888) >> 1;
+
+      // Ignore sign bit when indexing into LUT
+      uint32_t lut_idx = (src & 0x77777777);
+
+      // Signed is OR'd with 0x31203120 to find the correct value in the LUT
+      // (selects correct positive or negative candidate)
+      const uint32_t final_prmt_base = 0x32103210;
+
+      static constexpr uint32_t POS_INT8_REG1 = 0x03020100;  // [0, 1, 2, 3]
+      static constexpr uint32_t POS_INT8_REG2 = 0x07060504;  // [4, 5, 6, 7]
+      static constexpr uint32_t NEG_INT8_REG1 = 0xFBFAF9F8;  // [-8,-7,-6,-5]
+      static constexpr uint32_t NEG_INT8_REG2 = 0xFFFEFDFC;  // [-4,-3,-2,-1]
+
+      const int iters = std::max(PackedResultType::kElements / 4, size_t(1));
+      CUTLASS_PRAGMA_UNROLL
+      for (int ii = 0; ii < iters; ++ii, lut_idx >>= 16, sign >>= 16) {
+        uint32_t final_prmt_idx = final_prmt_base | sign;
+
+        // This uses a look up table to convert packed int4s to packed fp8s,
```

Comment:
This says `fp8s` but should say `int8s`?

Reply:
good catch, done!